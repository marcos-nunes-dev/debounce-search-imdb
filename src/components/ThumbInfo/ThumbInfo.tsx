import React, { ReactElement } from 'react';
import { Box } from 'components/Box';
import { Text } from 'components/Text';

interface Props {
    key?: string;
    image: string | null;
    title?: string;
    onClick?: any;
    alt?: string;
    width?: string;
    height?: string;
}

export function ThumbInfo({
    image,
    title,
    alt,
    width = '200px',
    height = '200px',
    ...rest
}: Props): ReactElement {
    return (
        <Box
            {...rest}
            display='flex'
            alignItems='center'
            flexDirection='column'
            mx='10px'
            mb='10px'
        >
            <a>
                <Box
                    borderRadius='100%'
                    role='img'
                    aria-label={alt || title || 'image'}
                    background={
                        image
                            ? `url(${image})no-repeat center center /cover`
                            : '#eaeaea'
                    }
                    width={width}
                    height={height}
                    data-testid='thumb-image'
                />
            </a>
            {title && (
                <Text
                    as='h2'
                    color='secondary'
                    fontWeight='bold'
                    fontSize='20px'
                >
                    {title}
                </Text>
            )}
        </Box>
    );
}
