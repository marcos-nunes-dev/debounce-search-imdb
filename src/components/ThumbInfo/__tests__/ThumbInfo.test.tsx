/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from 'tests';
import { ThumbInfo } from '../ThumbInfo';

describe('<ThumbInfo />', () => {
    it('should render ThumbInfo', () => {
        const { getByText } = render(
            <ThumbInfo
                image={null}
                title='Jhon Doe'
                alt='hi! im jhon doe'
                width='100px'
                height='100px'
            />
        );
        expect(getByText("Jhon Doe")).toBeTruthy();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <ThumbInfo
                image={null}
                title='Jhon Doe'
                alt='hi! im jhon doe'
                width='100px'
                height='100px'
            />
        );
        expect(container).toMatchSnapshot();
    });
});
