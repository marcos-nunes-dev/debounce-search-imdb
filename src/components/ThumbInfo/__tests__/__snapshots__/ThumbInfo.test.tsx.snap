// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThumbInfo /> should match snapshot 1`] = `
<div>
  <div
    class="Box-sc-1o1tflv-0 fEBBAJ"
    display="flex"
  >
    <a>
      <div
        aria-label="hi! im jhon doe"
        class="Box-sc-1o1tflv-0 ntrYP"
        data-testid="thumb-image"
        height="100px"
        role="img"
        width="100px"
      />
    </a>
    <h2
      class="Box-sc-1o1tflv-0 Text__TextStyled-sc-hojgsa-0 jTFqYA dfiLoS"
      color="secondary"
      font-size="20px"
      font-weight="bold"
    >
      Jhon Doe
    </h2>
  </div>
</div>
`;
